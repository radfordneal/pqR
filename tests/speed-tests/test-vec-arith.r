source("time.r")
 
a <- seq(1,2,length=1000)
b <- seq(3,4,length=1000)
s <- seq(5,6,length=4)
n <- 500000

test.name <- "vec-arith.+"
f <- test.cmp (function () { for (i in 1:n) r <- 1+a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a+1; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a+b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a+s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- s+a; r})
sys.time(r <- f()); print(sum(r))

test.name <- "vec-arith.-"
f <- test.cmp (function () { for (i in 1:n) r <- 1-a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a-1; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a-b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a-s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- s-a; r})
sys.time(r <- f()); print(sum(r))

test.name <- "vec-arith.*"
f <- test.cmp (function () { for (i in 1:n) r <- 3*a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a*3; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a*b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a*s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- s*a; r})
sys.time(r <- f()); print(sum(r))

test.name <- "vec-arith./"
f <- test.cmp (function () { for (i in 1:n) r <- 3/a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a/3; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a/b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a/s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- s/a; r})
sys.time(r <- f()); print(sum(r))
