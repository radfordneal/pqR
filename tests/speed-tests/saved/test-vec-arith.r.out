
---- vec-arith.+ 

Time:
[1] 2500
Time:
[1] 2500
Time:
[1] 5000
Time:
[1] 7000
Time:
[1] 7000

---- vec-arith.- 

Time:
[1] -500
Time:
[1] 500
Time:
[1] -2000
Time:
[1] -4000
Time:
[1] 4000

---- vec-arith.* 

Time:
[1] 4500
Time:
[1] 4500
Time:
[1] 5333.5
Time:
[1] 8250.417
Time:
[1] 8250.417

---- vec-arith./ 

Time:
[1] 2079.612
Time:
[1] 500
Time:
[1] 424.6279
Time:
[1] 273.9751
Time:
[1] 3812.414
